### PR TITLE
Bug/issue 322 trade widget

### LINKED ISSUE #322: trade widget bugs
- [ ] switching between market to limit - carry the user input value
flow:

- open market
- select from/to tokens
- input value in the from token field
- switch to limit view
expected: keep the input value when switching to the limit view.

- [ ] CTA text to connect a wallet
on market, we have a logical flow that shows "connect a wallet" text on the CTA if the user did not connect a wallet.
however, on limit, the text is "trade" and doesnt follow the same logic.
expected:
limit and market should follow the same CTA text logic when user wallet not connected, tokens are not selected and no input amount.

### PR DESCRIPTION
**Pull request by adarshram**
fixes #322 

image
image
market and limit input values carry over . 
There is a known issue i'm not able to fix . if you enter from the limit widget with a from value to market widget the you recieve column is not getting triggered . this is due to the fromDebounce value being false and  its triggered from the TokenInputField element . 
Example screen shot 
image

### CODE REVIEW

File: src/elements/swapLimit/SwapLimit.tsx [hunk 1]
```diff
@@ -112,7 +118,7 @@ export const SwapLimit = ({
         setFromAmountUsd(usdAmount);
       }
     },
-    [fromToken]
+    [fromToken,setFromAmount,setFromAmountUsd]
```
Comment (zavelevsky) [inline src/elements/swapLimit/SwapLimit.tsx:121]: please run prettier

Comment (adarshram) [reply src/elements/swapLimit/SwapLimit.tsx]: done .

Comment (adarshram) [reply src/elements/swapLimit/SwapLimit.tsx]: image
this issue seems to be happening after i merged with main .this only happens when you select a receive token and then swap from market to limit . if its in 'select a token' mode its fine

Comment (zavelevsky) [reply src/elements/swapLimit/SwapLimit.tsx]: Thanks, it was opened today - #437

Comment (adarshram) [reply src/elements/swapLimit/SwapLimit.tsx]: shall i fix that in this ticket . i figured the issue . or should i create a seperate branch for it ?

**Comment by adarshram** [comment]
Fixed the debounce issue mentioned in the ticket as well . Just added the fromAmount to the debounce and seems like it fixes the issue (instead of doing useDebounce(''))

**Comment by zavelevsky** [comment]
Note that in the issue (#322) there's also this requirement:
CTA text to connect a wallet
Did you address it?

**Comment by adarshram** [comment]
> Note that in the issue (#322) there's also this requirement: CTA text to connect a wallet Did you address it?

no i havent addressed this .

File: src/elements/modalDuration/modalDuration.tsx [hunk 2]
```diff
@@ -28,23 +28,23 @@ export const ModalDuration = ({
 
   const daysItems: DurationItem[] = Array.from(
     { length: 31 },
-    (index: number) => ({
+    (_,index: number) => ({
```
Comment (zavelevsky) [inline src/elements/modalDuration/modalDuration.tsx:31]: this fix was done already and it causes you conflicts. Also, again, please set prettier to automatically format please.

Comment (adarshram) [reply src/elements/modalDuration/modalDuration.tsx]: i resolved the conflict now ..

**Comment by zavelevsky** [comment]
@adarshram  I will merge your PR to another branch and create a PR from there. I have to do it because of a cloudflare limitation that doesn't allow me to generate preview links from an external PR. I want to share a preview link with our Product manager so that he can review your fix. So thanks for the contribution - this is just a technical step and then your code will be merged to main in this somewhat over-complex manner ¯\_(ツ)_/¯

**Comment by adarshram** [comment]
Awesome thanks I'm learning a lot from this . Will be fixing more over
time after learning some more react programming

**Comment by zavelevsky** [comment]
> 

Nice! I'd like that and I'm glad to hear.

**Review by zavelevsky — APPROVED** [review]
LGTM

**Comment by zavelevsky** [comment]
@adarshram issue (I'll fix over the new PR) - button text is correct but the button is disabled

**Comment by adarshram** [comment]
Okay alright . Will take a look tomorrow

**Pull request merged**